commands/snapshotcheckercommand.php

```diff
@@ -0,0 +1,119 @@
+            ->setDescription("Checks, if all snapshots of a server have been done and send success notification.");
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (is_readable('config.ini'))
+        {
+            $this->config = parse_ini_file('config.ini', true);
+            if (!isset($this->config['api']['user']) || !isset($this->config['api']['password'])) throw new Exception("No user or no password configured to connect ProfitBricks!");
+        }
+        else throw new Exception("Error during reading config.ini!");
+
+        $profitBricksApi = new ProfitBricksApi();
+        $profitBricksApi->setUserName($this->config["api"]["user"]);
+        $profitBricksApi->setPassword($this->config["api"]["password"]);
+        /** @var Snapshot[] $snapshots */
+        $snapshots = $profitBricksApi->snapshots();
+
+        // read from snapshot status file, which is created (and overwritten) by snapshot:autoCreate
+        $fileContent = file_get_contents(getcwd()."/checker.sav");
+        if ($fileContent===false) throw new Exception("Could not read file 'checker.sav'!");
+        $virtualMachineState = unserialize($fileContent);
+        if (count($virtualMachineState)>0) {
+            $io = new SymfonyStyle($input, $output);
+            $io->title("Snapshot checker  ".date("d.m.Y H:i:s"));
+            $tableRows = array();
+            $this->readVirtualMachines($profitBricksApi);
+            foreach ($virtualMachineState as $virtualMachineId => $snapshotStates) {
+                $valueChanged = false;
+                $available = true;
+                $snapshotDates = "";
+                foreach ($snapshotStates as $snapshotId => $snapshotState) {
+                    if (array_key_exists($snapshotId, $snapshots) && $snapshotState != $snapshots[$snapshotId]->state) {
+                        $virtualMachineState[$virtualMachineId][$snapshotId] = $snapshots[$snapshotId]->state;
+                        $valueChanged = true;
+                    }
+                    if ($virtualMachineState[$virtualMachineId][$snapshotId] != "AVAILABLE") $available = false;
+                    else if (array_key_exists($snapshotId, $snapshots)) $snapshotDates .= $snapshots[$snapshotId]->createdDate->format("Y-m-d H:i") . "; ";
+                }
+                if ($valueChanged && $available) {
+                    $action = "mail sent!";
+                    $this->sendMailForSnapshotVMs($virtualMachineId, $this->virtualMachines[$virtualMachineId]->name, $snapshotDates);
+                } else $action = "nothing to do!";
+                $tableRows[] = array($this->virtualMachines[$virtualMachineId]->name, $action);
+            }
+
+            $io->table(array("Server", "Action"), $tableRows);
+
+            file_put_contents(getcwd() . "/checker.sav", serialize($virtualMachineState));
+        }
+    }
+
+    /**
+     * @param $_virtualMachineId
```

Comment:
Please add at least the types of the input parameters.
So no one has to look up in the code that for example $_backupDates cannot be an array.